crates/red_knot_python_semantic/resources/mdtest/assignment/augmented.md

```diff
@@ -148,6 +147,37 @@ else:
     f = 42.0
 f += 12
 
-# TODO(charlie): This should be `str | float`.
-reveal_type(f)  # revealed: @Todo
+reveal_type(f)  # revealed: str | float
+```
+
+## Partially bound target union with `__add__`
+
+```py
+def bool_instance() -> bool:
+    return True
+
+flag = bool_instance()
+
+class Foo:
+    def __add__(self, other: int) -> str:
+        return "Hello, world!"
+    if bool_instance():
+        def __iadd__(self, other: int) -> int:
+            return 42
+
+
+class Bar:
+    def __add__(self, other: int) -> bool:
+        return True
+
+    def __iadd__(self, other: int) -> float:
+        return 42.0
+        
+if flag:
+    f = Foo()
+else:
+    f = Bar()
+f += 12
+
+reveal_type(f)  # revealed: int | str | float
```

Comment:
nit: I think this test will be clearer if you use a type other than `bool` here, since `bool` is a subtype of `int` and thus gets simplified out of the eventual union, which is a detail that's kind of extraneous to this test -- the point here is that all four types should be possible. I suggest `bytes`:

```suggestion
    def __add__(self, other: int) -> bytes:
        return True

    def __iadd__(self, other: int) -> float:
        return 42.0
        
if flag:
    f = Foo()
else:
    f = Bar()
f += 12

reveal_type(f)  # revealed: int | str | float | bytes
```

(order might be wrong on this assertion)

Reply:
Thanks!